pkg/pki/ca/controller/secret.go

```diff
@@ -214,6 +224,26 @@ func (sc *SecretController) scrtDeleted(obj interface{}) {
 	sc.upsertSecret(saName, scrt.GetNamespace())
 }
 
+func (sc *SecretController) generateKeyAndCert(saName string, saNamespace string) ([]byte, []byte, error) {
+	id := fmt.Sprintf("%s://cluster.local/ns/%s/sa/%s", ca.URIScheme, saNamespace, saName)
+	options := ca.CertOptions{
+		Host:       id,
+		RSAKeySize: keySize,
+	}
+
+	csrPEM, keyPEM, err := ca.GenCSR(options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	certPEM, err := sc.ca.Sign(csrPEM)
```

Comment:
I share the same feeling and I can refactor in a followup PR. 

One proposal is to move `GenCSR` and `GenCert` into a separate `x509` package. Then we'll have `sc.ca.Sign` and `x509.GenCSR`. Do you think this is less confusing?